app/src/main/java/co/tinode/tindroid/db/MessageDb.java

```diff
@@ -301,6 +301,53 @@ static Cursor queryDeleted(SQLiteDatabase db, long topicId, boolean hard) {
         return db.rawQuery(sql, null);
     }
 
+    /**
+     * Find the latest missing range of messages for fetching from the server.
+     *
+     * @param db database to select from;
+     * @param topicId Tinode topic ID (topics._id) to select from;
+     * @return range of missing IDs if found null if either all messages are present or no messages are found.
+     */
+    static MsgRange getNextMissingRange(SQLiteDatabase db, long topicId) {
+        int high = 0;
+        final String sqlHigh = "SELECT MAX(m1." + COLUMN_NAME_SEQ + ") AS missing" +
+                " FROM " + TABLE_NAME + " AS m1" +
+                " LEFT JOIN " + TABLE_NAME + " AS m2" +
+                " ON m1." + COLUMN_NAME_SEQ + "=IFNULL(m2." + COLUMN_NAME_HIGH + ", m2." + COLUMN_NAME_SEQ + "+1)" +
+                    " AND m2." + COLUMN_NAME_TOPIC_ID + "=?" +
+                " WHERE m2." + COLUMN_NAME_SEQ + " IS NULL" +
```

Comment:
Is it ever possible?
Insertions set `seq` to 0:
https://github.com/tinode/tindroid/blob/devel/app/src/main/java/co/tinode/tindroid/db/SqlStore.java#L286

Reply:
Yes, it's a left join. NULL is returned when the `ON` does not find a corresponding row.